app/models/manageiq/providers/vmware/cloud_manager/orchestration_service_option_converter.rb

```diff
@@ -1,11 +1,79 @@
 module ManageIQ::Providers
   class Vmware::CloudManager::OrchestrationServiceOptionConverter < ::ServiceOrchestration::OptionConverter
+    include Vmdb::Logging
+
     def stack_create_options
-      {
+      options = {
         :deploy  => stack_parameters['deploy'] == 't',
-        :powerOn => stack_parameters['powerOn'] == 't',
-        :vdc_id  => @dialog_options['dialog_availability_zone']
+        :powerOn => stack_parameters['powerOn'] == 't'
       }
+      options[:vdc_id] = @dialog_options['dialog_availability_zone'] unless @dialog_options['dialog_availability_zone'].blank?
+
+      options.merge!(customize_vapp_template(collect_vm_params))
+    end
+
+    private
+
+    # customize_vapp_template will prepare the options in a format suitable for the fog-vcloud-director.
+    # This mainly results in creating two top level objects in a hash. The :InstantiationParams, contains
+    # a single :NetworkConfig element with the array of all references to the networks used in the
+    # deployed vApp. We are using IDs here and let fog library create concrete HREFs that are required
+    # by the vCloud director. The second object is an array of source items. Each source item references
+    # a single VM from the vApp template, customises its name and optionally sets network info.
+    def customize_vapp_template(vm_params)
+      network_config = {}
+
+      source_vms = vm_params.collect do |vm_id, vm_opts|
+        src_vm = { :vm_id => "vm-#{vm_id}" }
+        src_vm[:name] = vm_opts["instance_name"] if vm_opts.key?("instance_name")
+
+        network_id = vm_opts["vdc_network"]
+        unless network_id.nil?
+          # Create new network config if it hasn't been created before.
+          network_config[network_id] = {
+            :networkName => network_id,
+            :networkId   => network_id,
+            :fenceMode   => "bridged"
+          } unless network_config.key?(network_id)
+
+          # Add network configuration to the source VM.
+          src_vm[:networks] = [
```

Comment:
It has to be an array of objects. Since all elements are `:key => value`, this implicitly creates an object. Would you prefer that I do it explicitly, so that I create it like

```
          src_vm[:networks] = [
            {
              :networkName             => network_id,
              :IsConnected             => true,
              :IpAddressAllocationMode => "DHCP"
            }
          ]
```